### ISSUE TITLE
Remove `wordpress-components` from frontend components

### ISSUE THREAD

**Issue by senadir** [woothemes/woocommerce#43652]
Using @wordpress/components is not supported and is not in the use cases, it causes significant performance degradation to Checkout, even thou we try to work around it by tree shaking, it's still not a good strategy.

This is a list of all components that we're using from @wordpress/components, removing them all will take some time and effort but is doable.

- SelectControl: [Ariakit Select](https://ariakit.org/components/select)
- Button: Let's build this.
- ComboboxControl: [Ariakit Combobox.](https://ariakit.org/components/combobox)
- FormTokenField: Might need to build, but this is also only used in one place, Attribute Filter Block, @gigitux and @albarin would that still be needed given the work you're doing on newer filter blocks?
- Radio: [Ariakit Radio](https://ariakit.org/components/radio)
- RadioGroup: [Ariakit Radio](https://ariakit.org/components/radio)
- SlotFillProvider: Either build this from scratch (doable), or contribute upstream to move Slots away from the component library to something else. Should note that we're using the bubblesVirtually version, which is newer.
- createSlotFill: Same as above.
- useSlot: Same as above.

Using `@wordpress/components` is not supported and is not in the use cases, it causes significant performance degradation to Checkout, even thou we try to work around it by tree shaking, it's still not a good strategy.

This is a list of all components that we're using from `@wordpress/components`, removing them all will take some time and effort but is doable.

**Comment by nerrad** [comment]
> `FormTokenField`

I think @dinhtungdu  or @samueljseay could help answer whether that is needed for the new filter blocks.

**Comment by nerrad** [comment]
Regarding Slot and Fill related components, you should check to see if there's something from the WP Interactivity API that can be utilized. At the very least, even if it doesn't have that currently, it could be contributed to the project?

**Comment by samueljseay** [comment]
> Using @wordpress/components is not supported and is not in the use cases

Just wanted to say massive +1 for this.

> I think @dinhtungdu or @samueljseay could help answer whether that is needed for the new filter blocks.

I'm excited to say that we've not been using FormTokenField or anything else from `@wordpress/components`.

The challenge (although surmountable) is that to create a reusable component that is server rendered but also has a "front-end editor side rendered" component means we have to build a lot of ui components again from scratch. For front-end filters we did this with [Dropdown](https://github.com/woocommerce/woocommerce/blob/1d0a00741ae45c4ef0a48f35f96704e0c8c46b8b/plugins/woocommerce/src/Blocks/InteractivityComponents/Dropdown.php#L10) and [CheckboxList](https://github.com/woocommerce/woocommerce/blob/1d0a00741ae45c4ef0a48f35f96704e0c8c46b8b/plugins/woocommerce/src/Blocks/InteractivityComponents/CheckboxList.php#L10). There is still more refinement to this approach needed, and it will get easier in future versions of WP because the server rendering APIs available will greatly increase. Sorry if this is a ramble but just some context.

TLDR; Yes please lets get rid of `FormTokenField`, and migrate to these new server rendered components we introduced. And if they're lacking features please let me know if I can enhance them, or of course contribute additional functionality.


> Regarding Slot and Fill related components, you should check to see if there's something from the WP Interactivity API that can be utilized. At the very least, even if it doesn't have that currently, it could be contributed to the project?

Interesting.. I would be happy to start a discussion about this with the interactivity team. I don't think anything exists currently and its a big question mark for me how that would work in an interactivity context. We can do things server side like nesting interactivity blocks inside each other, but thats a bit different from the way slot fill works. I think we really need to look over existing uses of SlotFill to ask how interactivity API can help with those use cases, because I'm not convinced SlotFill is the right approach within this framework

**Comment by samueljseay** [comment]
~~Another question aimed at clarifying and avoiding wasting time building the wrong thing:~~

~~Do we, going forward, need *any* **React-based front-end components**? Or can we move them all over to interactivity API straight away? It would be my preference to go to interactivity API for everything listed here (as previously mentioned I need to do some investigation regarding slot fill)~~

~~Additionally, I must mention / ask:~~

~~If we do go interactivity, then ariakit makes no sense at all right? its completely unstyled solution. We'd be better off building components from scratch or perhaps there are other frameworks or paradigms for building accessible HTML that could be utilized but as far as I understand ariakit is purely a react components based framework?~~

EDIT:

I didn't realize there was more nuance here. Checkout/Cart are complex React apps so migrating them to interactivity is not feasible.

I will raise this point for thought though (copy paste from other discussions):

"for Woo, I think we have a departure in 2 directions for front-end components now that we’re using the Interactivity API. It doesn’t make sense for us to mimic the markup and rendering of existing wordpress components in server side rendered interactivity API blocks. I tried it and it wasn’t worth it. As I mentioned, for example, FormTokenField is just way more markup than we need.

And that raises another issue for me. If in checkout and other React-based frontends we are using the same UI paradigms as in server side rendered blocks then unless we can somehow share the markup and styling we’ll need to maintain 2 sets of components which will make theming harder and general maintenance (such as style tweaks across both approaches) more difficult.

I haven’t spent the time to think in detail about this but I wonder if there is a better approach for Woo front-ends that would allow us to share components across these 2 paradigms. For example in the past I’ve seen front-end UI frameworks that were a basic style reset and some opinionated accessible HTML markup examples for each component. afaik Ariakit wouldn’t be as good for that kind of approach right? It’s strictly React components?

The other grain of salt here is how much all of this really matters. How many f/e components share UI paradigms and need to be consistent across both blocks and checkout? I haven’t quantified the answer to that, so I’m raising these thoughts for further discussion."

**Comment by wavvves** [comment]
Seems like SelectControl is used only by an unused experimental block soon to be removed in https://github.com/woocommerce/woocommerce/issues/44503
If so, for now we do not need to worry for a replacement 👍🏼

**Comment by opr** [comment]
After we worked on removing/updating some of these components (see [Epic](https://github.com/woocommerce/woocommerce/issues/45571)) the updated list looks like this:

- [x] ~`SelectControl`: [Ariakit Select](https://ariakit.org/components/select)~ - Not needed as per https://github.com/woocommerce/woocommerce/issues/46061
- [x] `Button`: Let's build this. - done in https://github.com/woocommerce/woocommerce/pull/45974/
- [ ] `ComboboxControl`: [Ariakit Combobox.](https://ariakit.org/components/combobox) see https://github.com/woocommerce/woocommerce/issues/46357
- [ ] ~`FormTokenField`: Might need to build, but this is also only used in one place, Attribute Filter Block, @gigitux and @albarin would that still be needed given the work you're doing on newer filter blocks?~
- [x] `Radio`: [Ariakit Radio](https://ariakit.org/components/radio) - done in https://github.com/woocommerce/woocommerce/pull/45974/
- [x] `RadioGroup`: [Ariakit Radio](https://ariakit.org/components/radio) - done in https://github.com/woocommerce/woocommerce/pull/45974/

The last 3 items were not re-written but we did update them in https://github.com/woocommerce/woocommerce/pull/47105 (see [reason](https://github.com/woocommerce/woocommerce/issues/47106)) so we can mark them as complete as well.
- [x] `SlotFillProvider`: Either build this from scratch (doable), or contribute upstream to move Slots away from the component library to something else. Should note that we're using the `bubblesVirtually` version, which is newer.
- [x] `createSlotFill`: Same as above.
- [x] `useSlot`: Same as above.

**Comment by opr** [comment]
Since only one outstanding task exists for this issue (https://github.com/woocommerce/woocommerce/issues/47757) I will close this. 

Progress can be tracked on https://github.com/woocommerce/woocommerce/issues/4775
